elogia_sale/models/elogia_sale.py

```diff
@@ -0,0 +1,811 @@
+    currency_id = fields.Many2one('res.currency', string="Currency")
+    clicks = fields.Float('Click/Lead/Sale', required=True, default=1)
+    amount_unit = fields.Float('Amount total', tracking=1)
+    amount_currency = fields.Float('Amount currency', tracking=1)
+    consume = fields.Float('Consume', tracking=1)
+    consume_currency = fields.Float('Consume currency', tracking=1)
+    client_id = fields.Many2one('res.partner', 'Client', tracking=1)
+    percentage_fee = fields.Float('% Fee', tracking=1)
+    check_change = fields.Boolean('Change?')
+    count_invoice_sale = fields.Integer('Invoice sale', compute='get_count_models')
+    count_invoice_purchase = fields.Integer('Invoice vendor', compute='get_count_models')
+    count_sale = fields.Integer('Sales', compute='get_count_models')
+    # count_picking = fields.Integer('Picking S', compute='get_count_models')
+    count_purchase = fields.Integer('Purchase', compute='get_count_models')
+    count_move = fields.Integer('Move', compute='get_count_models')
+    control_line_ids = fields.One2many('control.line.supplier', 'control_id', 'Control lines')
+    type_invoice = fields.Selection(selection=[
+        ('sale', 'Sales'),
+        ('account', 'Accounting')
+    ], string='Billing type', tracking=1)
+    team_id = fields.Many2one('crm.team', 'Sale team', tracking=1)
+    order_ids = fields.One2many('sale.order', 'control_id', 'Sales')
+    invoice_ids = fields.One2many('account.move', 'control_id', 'Invoices')
+    purchase_ids = fields.One2many('purchase.order', 'control_id', 'Purchase')
+
+    def set_pending(self):
+        for record in self:
+            record.state = 'pending'
+
+    def set_draft(self):
+        for record in self:
+            record.state = 'draft'
+
+    def set_cancel(self):
+        for record in self:
+            record.state = 'cancel'
+
+    def generate_sale(self):
+        for record in self:
+            pass
+
+    @api.onchange('campaign_id')
+    def onchange_campaign_id(self):
+        if self.campaign_id:
+            self.project_id = self.campaign_id.project_id.id
+            self.currency_id = self.campaign_id.currency_id.id
+            self.company_id = self.campaign_id.company_id.id
+            self.client_id = self.campaign_id.client_id.id
+            self.percentage_fee = self.campaign_id.percentage_fee
+            self.type_invoice = self.campaign_id.type_invoice
+            self.campaign_line_id = False
+
+    @api.onchange('client_id')
+    def onchange_client_id(self):
+        if self.client_id and self.client_id.team_id:
+            self.team_id = self.client_id.team_id
+
+    @api.onchange('campaign_line_id')
+    def onchange_campaign_line_id(self):
+        if self.campaign_line_id:
+            self.percentage_fee = self.campaign_line_id.percentage_fee
+
+    def get_count_models(self):
+        for record in self:
+            record.count_invoice_sale = len([item for item in record.invoice_ids if item.move_type
```

Comment:
podrias probar algo asi a ver si te funciona?? en aras de que quede mas cuqui el code, solo como sugerencia :) 

len(record.invoice_ids.filtered(lambda l: l.move_type  in ['out_invoice', 'out_refund']))

Reply:
Optimo